typings/@chainlink/index.d.ts

```diff
@@ -14,7 +14,11 @@ declare module '@chainlink/types' {
     data: AdapterResponse
   }
 
-  export type Execute = (input: AdapterRequest, config?: any) => Promise<AdapterResponse>
+  export type Execute = (input: AdapterRequest, config: any) => Promise<AdapterResponse>
+  export type ExecuteWithDefaults = (
+    input: AdapterRequest,
+    config?: any,
+  ) => Promise<AdapterResponse>
```

Comment:
Just thinking out loud for now, but I think we should move more in this direction:

```suggestion
  export type Execute = (input: AdapterRequest) => Promise<AdapterResponse>
  export type ExecuteWithConfig = (input: AdapterRequest, config: any) => Promise<AdapterResponse>
  export type ExecuteFactory = (config?: any) => Execute
```

Then we would build:

```javascript
const getDefaults = (): Config => getConfig()
const makeExecute: ExecuteFactory = (config) => async (request) => execute(request, config)
const execute: Execute = makeExecute(getDefaults())
```